geopandas/geodataframe.py

```diff
@@ -1593,6 +1594,11 @@ def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwargs):
     def _constructor(self):
         return _geodataframe_constructor_with_fallback
 
+    def _constructor_from_mgr(self, mgr, axes):
+        if not any(isinstance(block.dtype, GeometryDtype) for block in mgr.blocks):
+            return pd.DataFrame._from_mgr(mgr, axes)
```

Comment:
This is essentially the same logic as what we have in `_geodataframe_constructor_with_fallback` right?

Reply:
Yes, that's the intent.
I suppose you could alternatively write this as something like
```
def _constructor_from_mgr(self, mgr, axes):
    df = GeoDataFrame._from_mgr(mgr, axes)
    return _geodataframe_constructor_with_fallback(df)
```
to make that more explicit (and/  or less confusing) - I suppose that's effectively what pandas is doing right now when it throws the warnings. 

I did also wonder if we now have enough `_constructor` variants where it makes sense to collect them together into a separate module but I think that would be a mess because of partially initalised imports.

Reply:
I think it is fine to leave it as is. In any case, your snippet above would pass `df` to the GeoDataFrame constructor again (inside `_geodataframe_constructor_with_fallback`), so that would be unnecessary for this case.

> though if the recommendation for a subclass to implement `_constructor_from_mgr` is to just call the subclasses `_constructor`, it seems like the process of splitting this out into its own method is superfluous

I think the problem is that pandas doesn't necessarily know that (and it also doesn't know if the subclass' `_from_mgr` will return a fully instantiated object)